src/utils/promise.ts

```diff
@@ -0,0 +1,16 @@
+export interface ComponentWithUnmountState {
```

Comment:
Have also considered inherit from a base class that will handle the `isComponentUnmounted` state? So you don't need to implement it for all classes. (Open question)

Reply:
I can try it but as far as I know, making a base class for inheritance considered as bad practice in reactjs.

Reply:
Then it is fine.